### PR TITLE
Initial import of KV-snapshot benchmark

### PR DESCRIPTION
**Pull request by tzdybal**
* General benchmark structure
* KV-store interface
* RocksDB store implementation
* Initial approach to plotting with Gnuplot

### CODE REVIEW

**Comment by pkieltyka** [comment]
https://github.com/cockroachdb/pebble is another very strong RocksDB-compat KV to try out. My prediction is it will be the best

**Comment by robert-zaremba** [comment]
how about merging this PR?

**Comment by robert-zaremba** [comment]
> https://github.com/cockroachdb/pebble is another very strong RocksDB-compat KV to try out. My prediction is it will be the best

@pkieltyka  - did you benchmark pebble with RocksDB or baddger using a batched updates?

**Comment by liamsi** [comment]
Can you link to the results in a comment here? The plots still seem empty.

**Comment by tzdybal** [comment]
> Can you link to the results in a comment here? The plots still seem empty.

I still wait for them to finish generating ;)

File: benchmark.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package main
+
+type KV interface {
```
Comment (tac0turtle) [inline benchmark.go:3]: I think this should be the tm-db interface. That we can test what is already in use and possibly how we can modify the interface to better suit the needs here.

Comment (tzdybal) [reply benchmark.go:3]: I created new interface for maximum flexibility (and no dependencies). For example I need snapshots, but don't need "Sync" version of functions.

Comment (tac0turtle) [reply benchmark.go:3]: makes sense. I would hope the sdk would remove tm-db so this along with my plans.

Comment (robert-zaremba) [reply benchmark.go:3]: Let's firstly have a PoC and validate the approach with benchmarks.

**Pull request merged**